tests/push.bats

```diff
@@ -145,6 +145,16 @@ load helpers
     skip "docker is not installed"
   fi
 
+  # Confirm that this is really truly docker, not podman. Pick one or more:
+  docker_version=$(docker --version)
+  if [[ $docker_version =~ podman ]]; then
+    skip "this test needs docker, not podman-docker"
+  fi
+  docker_daemon=$(ps auxww|grep docker|grep -v grep)
+  if [[ -z "$docker_daemon" ]]; then
+    skip "docker daemon is not running"
+  fi
```

Comment:
Tagging as WIP because I don't know which of these is better, or even if there's another option. (Testing for `/run/docker.sock` is not useful; I think `podman system service` creates that).

Reply:
If we're going to push to dockerd, the service needs to be running (the "pull-from-docker-daemon" from pull.bats checks for this), and the `docker` command that we use to check what it has needs to be one that actually talks to it, and the first `if` test here looks good to me for that.  So I'm thinking both of those tests need to check for both of those things.

Reply:
Done, and refactored into a common helper. I also rebased.